elytron/WFCORE-4144-default-ssl_context-server.adoc

```diff
@@ -0,0 +1,59 @@
+= WFCORE-4144 Default SSLContext Registration (Within Application Server)
+:author:            Darran Lofthouse
+:email:             darran.lofthouse@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+This is a relatively simple enhancement to ensure that as the server is started a JVM wide default SSLContext is registered for the use of any libraries within the application server that support the use of the default context.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFCORE-4144[WFCORE-4144]
+
+=== Related Issues
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Affected Projects or Components
+
+WildFly Core
+
+=== Other Interested Projects
+
+None
+
+== Requirements
+
+=== Hard Requirements
+
+At the root of the WildFly Elytron subsystem a new attribute `default-ssl-context` will be registered.  This will be a reference to resources providing the `org.wildfly.security.ssl-context` capability.
+
+As the services for the subsystem start a reference to the `SSLContext` instance will be obtained and will be set globally using the API https://download.java.net/java/early_access/jdk11/docs/api/java.base/javax/net/ssl/SSLContext.html#setDefault(javax.net.ssl.SSLContext)[SSLContext.setDefault(SSLContext)].
```

Comment:
Just to make sure. There is no option deployment could start before this service will start, right? In other words there is not possible couple of calls will use old SSLContext and then new one will be used.

Reply:
I still need to work a little on that area to check, I was trying to avoid using a service dependency so I could guarantee registration happens during Stage.RUNTIME.

I may still be able to use a delegate to avoid any race conditions.